providers/aws/resources/aws_ec2.go

```diff
@@ -933,6 +896,10 @@ func (a *mqlAwsEc2) gatherInstanceInfo(instances []ec2types.Reservation, imdsvVe
 				"tpmSupport":            llx.StringDataPtr(instance.TpmSupport),
 			}
 
+			if instance.MetadataOptions != nil {
+				args["httpEndpoint"] = llx.StringData(string(instance.MetadataOptions.HttpEndpoint))
```

Comment:
should we make these fields optional in the lr?

Reply:
i dont think we can make those optional in the LR, but i did ensure that they are nil if the metadata struct is also nil (setting them to llx.NilData`